### PR TITLE
Added documentation for testing

### LINKED ISSUE #320: Docs for public test cases
This is just a reminder, that we need documentation for all public test cases under `Sonata\BlockBundle\Test`.

Refs:
https://github.com/sonata-project/SonataBlockBundle/pull/310
https://github.com/sonata-project/SonataBlockBundle/pull/308

### PR DESCRIPTION
**Pull request by core23**
I am targetting this branch, because this is a documentation.

Closes #320

## Subject

Added a short documentation for the public test classes.

### CODE REVIEW

File: Resources/doc/reference/testing.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Testing
+=======
+
+Test Blocks
+~~~~~~~~~~~
+
+You can write unit tests for block services with the following code.
+
+.. code-block:: php
```
Comment (OskarStark) [inline Resources/doc/reference/testing.rst]: could you include the `CustomBlockService`

File: Test/AbstractBlockServiceTestCase.php [hunk 2]
```diff
@@ -55,6 +55,13 @@ protected function setUp()
         $this->blockContextManager = new BlockContextManager($blockLoader, $this->blockServiceManager);
     }
 
+    /**
+     * Create a mocked block service.
+     *
+     * @param BlockServiceInterface $blockService A block service
+     *
+     * @return BlockContextInterface
+     */
```
Comment (OskarStark) [inline Test/AbstractBlockServiceTestCase.php]: canyou please do the phpdoc change at least in a seperate commit? thank you @core23 👍

Comment (core23) [inline Test/AbstractBlockServiceTestCase.php]: Done

Comment (OskarStark) [inline Test/AbstractBlockServiceTestCase.php]: Thank you :+1:

**Comment by OskarStark** [comment]
LGTM

**Pull request merged**